ledger/apply/application.go

```diff
@@ -342,11 +342,7 @@ func ApplicationCall(ac transactions.ApplicationCallTxnFields, header transactio
 		if err != nil {
 			return
 		}
-		// No separate config for activating storage in AD because
-		// inner transactions can't be turned on without this change.
-		if balances.ConsensusParams().MaxInnerTransactions > 0 {
-			ad.ApplicationID = appIdx
-		}
+		ad.ApplicationID = appIdx
```

Comment:
Now, we always fill the AD with this useful fact (which is then recorded by using RecordAD) and we then remove it later based on protocol version, so it does not change old block evaluations.  This is more inline with how rewards AD fields were handled, and makes it easier to reason about code that wants to use this information.